src/Core/WinSWCore/Configuration/YamlConfiguration.cs

```diff
@@ -315,15 +325,29 @@ public class YamlDownload
 
         public class YamlFailureAction
         {
-            [YamlMember(Alias = "type")]
-            private SC_ACTION_TYPE type;
+            [YamlMember(Alias = "action")]
+            public string? FailureAction { get; set; }
 
             [YamlMember(Alias = "delay")]
-            private TimeSpan delay;
+            public string? FailureActionDelay { get; set; }
+
+            public SC_ACTION_TYPE Type
+            {
+                get
+                {
+                    SC_ACTION_TYPE actionType = this.FailureAction switch
+                    {
+                        "restart" => SC_ACTION_TYPE.SC_ACTION_RESTART,
+                        "none" => SC_ACTION_TYPE.SC_ACTION_NONE,
+                        "reboot" => SC_ACTION_TYPE.SC_ACTION_REBOOT,
+                        _ => throw new Exception("Invalid failure action: " + this.FailureAction)
```

Comment:
Don't throw `Exception`.

Reply:
Which value should be bound to actionType for a invalid input?

Reply:
Throw `InvalidDataException` ~~or `InvalidEnumArgumentException`~~ instead.